### PR TITLE
L1 normalization added.

### LINKED ISSUE #15824: L1_normalize

### PR DESCRIPTION
**Pull request by Suraj520**
Added L1 normalization and migrated it to experimental.
Closes #15824

### CODE REVIEW

**Comment by zaeemansari70** [comment]
Hi @Suraj520, 
I've seen you are still implementing the function in the PR, 
kindly do request a review by commenting on the PR once you are done! :slightly_smiling_face:

**Comment by Suraj520** [comment]
> Hi @Suraj520, I've seen you are still implementing the function in the PR, kindly do request a review by commenting on the PR once you are done! slightly_smiling_face

Hi @zaeemansari70 : I'm not implementing anything as of now. Can you please review :smiley:

**Review by zaeemansari70 — CHANGES_REQUESTED** [review]
@Suraj520 , as this is a fundamental function this has to be a primary implementation which means it has to be implemented in all the backends separately, also you have to add the tests for the function as well, I'd suggest you to see the docs for this. https://lets-unify.ai/docs/ivy/overview/deep_dive/function_types.html
One more thing, your PR also not linked to an issue, kindly link the PR to the issue as well :slightly_smiling_face: 
Thanks! :slightly_smiling_face:

**Comment by Suraj520** [comment]
@zaeemansari70 - I have implemented L1 normalization across all backends, Migrated it to experimental and written the unit tests. Kindly review :smile:

**Review by zaeemansari70 — CHANGES_REQUESTED** [review]
Awesome @Suraj520 !

I've requested some more changes :slightly_smiling_face:  

1. As this is a backend implementation, you should also add the `array` and `container` instance methods, kindly add those.
2. Please resolve the merge conflicts too :slightly_smiling_face: 

Thanks! :slightly_smiling_face:

**Comment by Suraj520** [comment]
> Awesome @Suraj520 !
> 
> I've requested some more changes slightly_smiling_face
> 
> 1. As this is a backend implementation, you should also add the `array` and `container` instance methods, kindly add those.
> 2. Please resolve the merge conflicts too slightly_smiling_face
> 
> Thanks! slightly_smiling_face

Hey @zaeemansari70 

I've resolved the merge conflicts but I am not sure how and where to add the array and container instance methods. Can you please elaborate upon how and where it should be implemented with a reference example ?

Thanks !

**Review by zaeemansari70 — COMMENTED** [review]
Hi @Suraj520 ,
So there is a folder known as `data_classes` inside ivy,  the path is `ivy/data_classes`
You will see the `array` and `container` folders there, you have to add these in the experimental folder for the respective files, you'll get an idea how to add them by looking at other implementations.
It's fairly simple :slightly_smiling_face:

**Comment by Suraj520** [comment]
> Hi @Suraj520 , So there is a folder known as `data_classes` inside ivy, the path is `ivy/data_classes` You will see the `array` and `container` folders there, you have to add these in the experimental folder for the respective files, you'll get an idea how to add them by looking at other implementations. It's fairly simple slightly_smiling_face

@zaeemansari70  : Done ! Please review :smile:

**Comment by zaeemansari70** [comment]
I am manually linking your PR to an issue, kindly do not forget to do this in the future.

File: ivy_tests/test_ivy/test_functional/test_experimental/test_nn/test_norms.py [hunk 1]
```diff
@@ -1,10 +1,40 @@
 # global
-from hypothesis import strategies as st
+from hypothesis import given, strategies as st
 
 # local
 import ivy_tests.test_ivy.helpers as helpers
 from ivy_tests.test_ivy.helpers import handle_test
 
+@handle_test(
+    fn_tree="functional.ivy.l1_normalize",
```
Comment (zaeemansari70) [inline ivy_tests/test_ivy/test_functional/test_experimental/test_nn/test_norms.py:9]: Sorry I missed this, this should be called from experimental Ivy

**Review by zaeemansari70 — CHANGES_REQUESTED** [review]
One last change :slightly_smiling_face:

File: ivy_tests/test_ivy/test_functional/test_experimental/test_nn/test_norms.py (hunk 1 above)
Comment (Suraj520) [reply ivy_tests/test_ivy/test_functional/test_experimental/test_nn/test_norms.py]: Thanks for pointing this out ! Fixed it :smile: ! Kindly check

**Comment by Suraj520** [comment]
> lgtm



> @Suraj520 I am manually linking your PR to an issue, kindly do not forget to do this in the future.

Sure ! Noted :blush:

**Review by zaeemansari70 — APPROVED** [review]
As the tests are also passing, the changes lgtm! :+1:

**Comment by Suraj520** [comment]
> As the tests are also passing, the changes lgtm! +1

Thanks !

**Pull request merged**